whois-update/src/main/java/net/ripe/db/whois/update/generator/SourceGenerator.java

```diff
@@ -21,30 +28,69 @@ public class SourceGenerator extends AttributeGenerator {
 
     private final AuthoritativeResourceData authoritativeResourceData;
 
+    private final RpslObjectDao rpslObjectDao;
     private final CIString source;
     private final CIString nonAuthSource;
 
     @Autowired
     public SourceGenerator(final AuthoritativeResourceData authoritativeResourceData,
                            @Value("${whois.source}") final String source,
-                           @Value("${whois.nonauth.source}") final String nonAuthSource) {
+                           @Value("${whois.nonauth.source}") final String nonAuthSource,
+                           final RpslObjectDao rpslObjectDao) {
         this.authoritativeResourceData = authoritativeResourceData;
         this.source = ciString(source);
         this.nonAuthSource = ciString(nonAuthSource);
+        this.rpslObjectDao = rpslObjectDao;
     }
 
     @Override
     public RpslObject generateAttributes(RpslObject originalObject, RpslObject updatedObject, Update update, UpdateContext updateContext) {
-        switch (updatedObject.getType()) {
-            case AUT_NUM:
-            case ROUTE:
-            case ROUTE6:
-                return generateSource(updatedObject, update, updateContext);
-            default:
+        return switch (updatedObject.getType()) {
+            case AS_SET -> generateAsSetSource(updatedObject, update, updateContext);
+            case AUT_NUM, ROUTE, ROUTE6 -> generateSource(updatedObject, update, updateContext);
```

Comment:
we need a break statement?

Reply:
no because we are doing a return previously